src/modules/Packages.rb

```diff
@@ -2381,7 +2381,32 @@ def current_proposal
     #
     # @return [Boolean] changed ?
     def proposal_changed?
-      current_proposal != @cached_proposal
+      changed = current_proposal != @cached_proposal
+      log_proposal_diff if changed
+      changed
+    end
+
+    #
+    # Log changed proposal
+    #
+    def log_proposal_diff
```

Comment:
NP: This two methods looks a bit cryptic to me.

I would unify them in a single (and more specific) one, since `log_array_diff` is is only used here, what do you think?